Sources/TuistCore/Utils/OtoolController.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import RxSwift
+import TSCBasic
+import TuistSupport
+
+enum OtoolErrors: FatalError, Equatable {
+    case invalidOutput
+
+    var description: String {
+        switch self {
+        case .invalidOutput: return "The output is not in the format tuist expects"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .invalidOutput: return .abort
+        }
+    }
+}
+
+/// Otool controller protocol passed in initializers for dependency injection
+public protocol OtoolControlling {
+    /// Returns the paths to dlyb dependencies.
+    /// - Parameter path: Path to the given dependency
+    /// - Returns Array of dependencies returned by the command.
+    func dlybDependenciesPath(forBinaryAt path: AbsolutePath) throws -> Single<[AbsolutePath]>
+}
+
+/// OtoolController
+/// Used to find out against which dynamic libraries a certain binary is linked.
+public struct OtoolController: OtoolControlling {
+    private let system: Systeming
+
+    public init(system: Systeming = System.shared) {
+        self.system = system
+    }
+
+    public func dlybDependenciesPath(forBinaryAt path: AbsolutePath) throws -> Single<[AbsolutePath]> {
+        let arguments = ["otool", "-L", path.pathString]
+
+        return System.shared.observable(arguments)
+            .mapToString()
+            .collectAndMergeOutput()
+            .map { try OtoolOutput(filePath: path, output: $0) }
+            .map(\.paths)
+            .asSingle()
+    }
+
+    /// Given a framework path it returns the path to its binary.
+    /// - Parameter frameworkPath: Framework path.
+    func binaryPath(frameworkPath: AbsolutePath) -> AbsolutePath {
```

Comment:
Is this method use anywhere?

Reply:
Actually it isn't. I think I was using it in the first implementation.